### PR TITLE
Update tests/helpers.py and nginx template for node selection constraints

### PR DESCRIPTION
**Pull request by sidhant-agrawal**
Fixes: #1255 
Fixes: #1256 

Signed-off-by: Sidhant Agrawal <sagrawal@redhat.com>

### CODE REVIEW

File: tests/helpers.py [hunk 1]
```diff
@@ -117,6 +117,7 @@ def create_pod(
         raw_block_device (str): raw block device for the pod
         replica_count (int): Replica count for deployment config
         pod_name (str): Name of the pod to create
+        node_selector (dict): dict of key-value pair to be used for nodeSelector field
```
Comment (ramkiperiy) [inline tests/helpers.py]: Please add a example

Comment (sidhant-agrawal) [reply tests/helpers.py]: Done

Comment (sidhant-agrawal) [reply tests/helpers.py]: Done

**Comment by sidhant-agrawal** [comment]
The tier1 tests which failed in the first two runs passed separately in the last run

**Pull request merged**